### PR TITLE
Update schema for RootStoreStatus

### PR DESCRIPTION
**Pull request by dadrian**
This schema change is breaking with regard to types.

### CODE REVIEW

**Comment by dadrian** [comment]
We have a problem. We had apparently already defined an incorrect schema for this.

File: ztag/schema.py [hunk 1]
```diff
@@ -339,21 +339,27 @@ class CensysString(WhitespaceAnalyzedString):
...
-    "was_valid":Boolean(doc="was this certificate ever valid in this browser")
+root_store_status = SubRecord({
+    "valid":Boolean(doc="((has_trusted_path && !revoked && !blacklisted) || whitelisted) && !expired"),
+    "was_valid":Boolean(doc="True if the certificate is valid now or was ever valid in the past."),
+    "has_trusted_path":Boolean(doc="True if there exists a path from the certificate to the root store."),
+    "had_trusted_path":Boolean(doc="True if now or at some point in the past there existed a path from the certificate to the root store."),
+    "blacklisted":Boolean(doc="True if the certificate is explicitly blacklisted by some method than OneCRL / CRLSet. For example, a set of certificates revoked by Cloudflare are blacklisted by SPKI hash in Chrome."),
+    "whitelisted":Boolean(doc="True if the certificate is explicitly whitelisted, e.g. the set of trusted WoSign certificates Apple uses."),
+    "type":Boolean(doc="Indicates if the certificate is a root, intermediate, or leaf."),
+    "paths":ListOf(ListOf(IndexedBinary())),
```
Comment (dadrian) [inline ztag/schema.py]: Is this what we want? Should we instead of a `ListOf` single-element `SubRecord`'s?

File: ztag/schema.py [hunk 2]
```diff
@@ -339,21 +339,27 @@ class CensysString(WhitespaceAnalyzedString):
...
+    "in_revocation_set":Boolean(doc="True if the certificate is in the revocation set (e.g. OneCRL) associated with this root store."),
+    "parents":ListOf(IndexedBinary()),
```
Comment (dadrian) [inline ztag/schema.py]: Not sure which of these should have `doc`

**Comment by dadrian** [comment]
Comments on old version are still relevant, but I fixed the schema to use HexString() and Enum().

**Comment by zakird** [comment]
Has trusted path should probably become just trusted path because that's
how all other fields are named for the present tense case.

**Comment by dadrian** [comment]
@zakird Update. Do you think we should change that in the protobuf as well?

**Comment by zakird** [comment]
Probably, but I won't force the issue.

**Pull request merged**